src/profile-logic/marker-timing.js

```diff
@@ -159,7 +189,20 @@ export function getMarkerTimingAndBuckets(
       return a.bucket > b.bucket ? 1 : -1;
     }
     if (a.name === b.name) {
-      // Keep the original ordering if the names are the same.
+      // These 2 lines are for the same marker name.
+
+      // The instant markers need to come first.
+      // All instant markers for one given name are in the same marker timing
+      // line, and we know all lines have at least one marker, so we can safely
+      // use these conditions.
+      if (a.end[0] === a.start[0]) {
```

Comment:
note: I tried various things to avoid this ugly comparison:

* supporting `null` values makes the code in the marker chart's canvas uglier
* making the `end` property `null` makes the code both here and in the marker chart's canvas more complex. Also this would make that a line is always of one type, which possibly won't always be the case.
* we could add a boolean or string property to the marker timing `timingLineType: 'instant' | 'interval'` but this would make that a line is always of one type, which possibly won't always be the case (with the various reorganization strategies I want to try out).

So in the end this was the easiest solution, but maybe we can move to adding another property in the future if one line contains only markers of one type.

Reply:
I see, before reading I was also thinking that we can add something like `timingLineType` to the `MarkerTiming` object, but if it's not guaranteed to contain only one type all the time, we can keep this way. One question though, if we might have a line that doesn't always have only one type, how can we make sure that the first marker is going to be the type that we want?

Reply:
Mmm good remark, here we actually assume that the full line will be the same.
Maybe an extra property is a good idea then!